File: CS/Form1.cs
using Syncfusion.WinForms.Controls;
using Syncfusion.WinForms.DataGrid.Styles;
using System.Windows.Forms;
using Syncfusion.WinForms.DataGrid.Enums;
using Syncfusion.WinForms.DataGrid;

namespace GettingStarted
{   
    public partial class Form1 : Form
    {
        #region Constructor
        public Form1()
        {
            InitializeComponent();
            var data = new OrderInfoCollection();
            sfDataGrid.DataSource = data.OrdersListDetails;
        }

        #endregion 

        private void textBox1_TextChanged(object sender, System.EventArgs e)
        {
            this.sfDataGrid.SearchController.Search(this.textBox1.Text);
        }

        private void nextButton_Click(object sender, System.EventArgs e)
        {
            this.sfDataGrid.SearchController.FindNext(this.textBox1.Text);
            SetSelectedItem();
        }

        private void previousButton_Click(object sender, System.EventArgs e)
        {
            this.sfDataGrid.SearchController.FindPrevious(this.textBox1.Text);
            SetSelectedItem();
        }

        private void SetSelectedItem()
        {
            var rowIndex = this.sfDataGrid.SearchController.CurrentRowColumnIndex.RowIndex;
            var recordIndex = this.sfDataGrid.TableControl.ResolveToRecordIndex(rowIndex);
            this.sfDataGrid.SelectedIndex = recordIndex;
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports Syncfusion.WinForms.Controls
Imports Syncfusion.WinForms.DataGrid.Styles
Imports System.Windows.Forms
Imports Syncfusion.WinForms.DataGrid.Enums
Imports Syncfusion.WinForms.DataGrid

Namespace GettingStarted
	Partial Public Class Form1
		Inherits Form
		#Region "Constructor"
		Public Sub New()
			InitializeComponent()
			Dim data = New OrderInfoCollection()
			sfDataGrid.DataSource = data.OrdersListDetails
		End Sub

		#End Region 

		Private Sub textBox1_TextChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles textBox1.TextChanged
			Me.sfDataGrid.SearchController.Search(Me.textBox1.Text)
		End Sub

		Private Sub nextButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
			Me.sfDataGrid.SearchController.FindNext(Me.textBox1.Text)
			SetSelectedItem()
		End Sub

		Private Sub previousButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button2.Click
			Me.sfDataGrid.SearchController.FindPrevious(Me.textBox1.Text)
			SetSelectedItem()
		End Sub

		Private Sub SetSelectedItem()
			Dim rowIndex = Me.sfDataGrid.SearchController.CurrentRowColumnIndex.RowIndex
			Dim recordIndex = Me.sfDataGrid.TableControl.ResolveToRecordIndex(rowIndex)
			Me.sfDataGrid.SelectedIndex = recordIndex
		End Sub
	End Class
End Namespace
